### PR TITLE
cli code refactoring

### PR DESCRIPTION
**Pull request by satish153**

### CODE REVIEW

**Review by weiqiand — COMMENTED** [review]
1. The def target_wrapper function in tdiTableEntry.py should be moved to target specific, since prsr_id is target specific. For p4-driver, don't have this.
2. We might consider the pipe and gress_dir is PNA specific not core class specific.
3. You might consider to move the tdiTableEntry.py to target specific. since it will contains prsr_id (target specific).

File: tdi_python/tdiDefs.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
+from enum import Enum, auto
```
Comment (weiqiand) [inline tdi_python/tdiDefs.py:1]: Please put Intel copyright statement

File: tdi_python/tdiDefs.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    TDI_MATCH_TYPE_DEVICE = 0x80
+
+# tdi_match_type_core_e
+class KeyMatchType:
+    class KeyMatchEnumCore(Enum):
+        TDI_MATCH_TYPE_EXACT = KeyMatchEnum.TDI_MATCH_TYPE_CORE.value
+        TDI_MATCH_TYPE_TERNARY = auto()
+        TDI_MATCH_TYPE_LPM = auto()
+        TDI_MATCH_TYPE_RANGE = auto()
+        TDI_MATCH_TYPE_OPTIONAL = auto()
```
Comment (weiqiand) [inline tdi_python/tdiDefs.py:64]: TDI_PSA_MATCH_TYPE_OPTIONAL for PSA arch type

File: tdi_python/tdiTable.py [hunk 3]
```diff
@@ -80,17 +88,20 @@ def __init__(self, cintf, handle, info, info_handle):
...
+        # #Unify the table name for TDINode (command nodes)
+        # name_lowercase_without_dollar=self.name.lower().replace("$","")
+        # if self.table_type in ["PORT_CFG", "PORT_STAT", "PORT_HDL_INFO", "PORT_FRONT_PANEL_IDX_INFO", "PORT_STR_INFO"]:
+        #     self.name = "port.{}".format(name_lowercase_without_dollar)
+        # if self.table_type in ["PRE_MGID", "PRE_NODE", "PRE_ECMP", "PRE_LAG", "PRE_PRUNE", "PRE_PORT", "MIRROR_CFG"]:
+        #     self.name = name_lowercase_without_dollar
+        # if self.table_type in ["TM_PORT_GROUP_CFG", "TM_PORT_GROUP"]:
+        #     self.name = name_lowercase_without_dollar
+        # if self.table_type in ["SNAPSHOT", "SNAPSHOT_LIVENESS"]:
+        #     self.name = "{}".format(name_lowercase_without_dollar)
```
Comment (saynb) [inline tdi_python/tdiTable.py:104]: Can we remove commented code ?

File: tdi_python/tdicli.py [hunk 4]
```diff
@@ -306,16 +311,16 @@ def get_device(self):
...
-        sts = self.get_driver().tdi_target_get_value(target, self.target_type_map(target_type_str="direction"), byref(direction));
-        prsr_id = c_uint64(0);
-        sts = self.get_driver().tdi_target_get_value(target, self.target_type_map(target_type_str="prsr_id"), byref(prsr_id));
-        return "dev_id={} pipe={} direction={} prsr_id={}".format(dev_id.value, pipe_id.value, direction.value, prsr_id.value)
+    # def print_target(self, target):
+    #     dev_id = c_uint64(0);
+    #     sts = self.get_driver().tdi_target_get_value(target, self.target_type_map(target_type_str="dev_id"), byref(dev_id));
+    #     pipe_id = c_uint64(0);
+    #     sts = self.get_driver().tdi_target_get_value(target, self.target_type_map(target_type_str="pipe_id"), byref(pipe_id));
+    #     direction = c_uint64(0);
```
Comment (saynb) [inline tdi_python/tdicli.py:319]: Let's remove commented code

File: tdi_python/tdicli.py [hunk 5]
```diff
@@ -50,11 +51,15 @@
...
 class CIntfTdi:
+    target_type_cls = TargetType
 
-    def __init__(self, dev_id, table_cls, info_cls):
+    def __init__(self, dev_id, table_cls, info_cls, driver_path, utils_path):
+    #def __init__(self, dev_id, table_cls, info_cls):
         self._dev_id = dev_id
-        self._utils = CDLL(install_directory+'/lib/libtarget_utils.so', mode=RTLD_GLOBAL)
-        self._driver = CDLL(install_directory+'/lib/libdriver.so')
+        self._utils = CDLL(utils_path, mode=RTLD_GLOBAL)
```
Comment (weiqiand) [inline tdi_python/tdicli.py:59]: Why do we need to change this code? you means each target can def its own libtarget_utils.so?

File: tdi_python/tdicli.py [hunk 6]
```diff
@@ -2304,94 +2309,148 @@ def tdi_exit_handler():
...
+        pass
 
+'''
 def main():
     start_tdi(0, 1, os.getcwd(), [0])
 
 if __name__ == "__main__":
     main()
 
+'''
```
Comment (saynb) [inline tdi_python/tdicli.py:2456]: need to correct the file ending

**Review by saynb — APPROVED** [review]

File: tdi_python/tdicli.py [hunk 7]
```diff
@@ -50,11 +51,15 @@
...
-    def __init__(self, dev_id, table_cls, info_cls):
+    def __init__(self, dev_id, table_cls, info_cls, driver_path, utils_path):
+    #def __init__(self, dev_id, table_cls, info_cls):
         self._dev_id = dev_id
-        self._utils = CDLL(install_directory+'/lib/libtarget_utils.so', mode=RTLD_GLOBAL)
-        self._driver = CDLL(install_directory+'/lib/libdriver.so')
+        self._utils = CDLL(utils_path, mode=RTLD_GLOBAL)
+        self._driver = CDLL(driver_path)
+        # self._utils = CDLL(install_directory+'/lib/libtarget_utils.so', mode=RTLD_GLOBAL)
+        # self._driver = CDLL(install_directory+'/lib/libdriver.so')
```
Comment (weiqiand) [inline tdi_python/tdicli.py:62]: Please move the functions: _set_pipe, _set_direction, _set_parser to target specific (line 223-230).
The constructor: self._dev_tgt = self.TdiDevTgt(self._dev_id, 0, 0xff, 0xff) to target specific too.

**Review by weiqiand — COMMENTED** [review]
For the tdicli.py and tdiTableEntry.py still contains target specific for prsr_id and need to consider to move them to target specific.

File: tdi_python/tdiDefs.py (hunk 2 above)
Comment (satish153) [reply tdi_python/tdiDefs.py:84]: As discussed, we are having only two levels of hierarchy. target/arch independent code in TDI. Dependent code in target repos.

File: tdi_python/tdicli.py (hunk 5 above)
Comment (satish153) [reply tdi_python/tdicli.py:59]: It means that the targets can specify their own path to the libraries.

File: tdi_python/tdicli.py (hunk 7 above)
Comment (satish153) [reply tdi_python/tdicli.py:62]: Will remove them in next phase of refactoring work

File: tdi_python/tdicli.py (hunk 4 above)
Comment (satish153) [reply tdi_python/tdicli.py:310]: Done

File: tdi_python/tdicli.py (hunk 6 above)
Comment (satish153) [reply tdi_python/tdicli.py:2387]: Done

**Pull request merged**